### PR TITLE
305 bounded voronoi tessellations

### LINKED ISSUE #305: Extend the functionality of Bounded Voronoi Tessellations
How Voronoi Tesselations is currently hard coded. 

i.e the nodes added are put at a specified distance and at a specified location along the mesh edge. 

We should add member variables to allow the modification of these hard coded parameters.

### PR DESCRIPTION
**Pull request by jmosborne**
Merging in changes to how Bounded Voronoi Tessellations are calculated.

Specifically make it so you can 
 *set a edge length to ignore to match a cut off length (like used for an Off Lattice force).
 * Make it so that the bound changes with cell length to more accurately calculate cell size on the edge.

### CODE REVIEW

**Comment by jmosborne** [comment]
Looks like everything is passing. Can i get an approval @fcooper8472

**Comment by jmosborne** [comment]
Adding Gary as reviewer as I see Fergus is at conference this week. And Gary loves looking at my code!

File: crypt/test/data/SteadyStateCrypt/archive/cell_population_sim_at_time_150.arch [hunk 1]
```diff
@@ -19,4947 +19,4947 @@
...
-20 0 0 25 12 1.50000000000000000e+02 4294967295 0 0 0 0 0 5 1 0 1 49 12 CellEdgeData 1 0
```
Comment (mirams) [inline crypt/test/data/SteadyStateCrypt/archive/cell_population_sim_at_time_150.arch:22]: Not quite obvious to me why 
 * (i) this archive has needed to change, 
 * (ii) this alteration from `CellEdgeData` to `CellId` has occurred even if archive needs updating? Is this expected from some other work, and we managed backwards compatible archive loading until now?

**Review by mirams — APPROVED** [review]

**Comment by mirams** [comment]
I approved, but there was one comment to check :-) https://github.com/Chaste/Chaste/pull/306#discussion_r1746856361

File: crypt/test/data/SteadyStateCrypt/archive/cell_population_sim_at_time_150.arch (hunk 1 above)
Comment (jmosborne) [reply crypt/test/data/SteadyStateCrypt/archive/cell_population_sim_at_time_150.arch:22]: The archive changed as the member variables of MeshBasedCellPopulation changed so things wont load anymore I don't know why the reordering happened as its archiving magic.

Comment (jmosborne) [reply crypt/test/data/SteadyStateCrypt/archive/cell_population_sim_at_time_150.arch:22]: Looking a bit more i think the change to the archive (i.e the switch between CellIID ad CellEdgeData) happened earlier but didn't break the loading of the archive so didnt set anything off.  This time non of the numerical values have changed so all is fine. Im actually going to make an issue to discuss the ArchiveFormat as I'm not sure its a worthwhile test anymore ansd may want to look to replcace it. Or make it catch issues sooner.

**Pull request merged**